exercice2.php

```diff
@@ -2,6 +2,40 @@
 
 require_once('vendor/autoload.php');
 
-use Ipssi\Evaluation\Useless;
+use Ipssi\Evaluation\Document;
+use Ipssi\Evaluation\ELement;
+use Ipssi\Evaluation\Text;
+use Ipssi\Evaluation\Forme;
+use Ipssi\Evaluation\Image;
+
+$doc = new Document(255, 255, 255); // création d'un documment blanc.
+$elements[] = new Forme(60, 200, 255, 0, 0, 'Nuage'); // Forme de nuage rouge à 60x, 200y.
+$elements[] = new Text(349, 10, 0, 255, 0, 'Salut');// Text "Salut" vert à 349x, 10y.
+$elements[] = new Image(190, 100, "chemin/image.png");// image à 190x, 100y.
+
+foreach($elements as $element) {
+    $doc->newElement($element->getId());
+}
+
+$color=$doc->getCouleur();
+echo "\n\n";
+echo "Couleur du document : R = " . $color[0] . " | V = " . $color[1] . " | B = " . $color[2] . "\n\n";
+echo "Eléments dans le document :" . "\n\n";
+
+foreach($elements as $element) {
+    if( array_search($element->getId(), $doc->getElements())) {
+        echo "ID de l'élément : " . $element->getId() . "\n";
+        echo "Nom de l'élément : " . $element->getName() . "\n";
+
+        $position = $element->getPosition();
+        echo "Position : X = " . $position[0] . " | Y = " . $position[1] . "\n";
+
+        if($element instanceof Text || $element instanceof Text) {
```

Comment:
```suggestion
        if($element instanceof Text || $element instanceof Forme) {
```
Bien ! Avec une interface c'est mieux : 
```php
interface Colorable 
{
    public function getColor(): Color;
}

class Forme extends Element implements Colorable{}
class Text extends Element implements Colorable{}

if ($element instanceof Colorable) {
    echo "Couleur R = " . $element->getColor()->getRed() . " | V = " .  $element->getColor()->getGreen() . " | B= " .  $element->getColor()->getBlue() . "\n";
}
```